connectors/sources/aws.py

```diff
@@ -3,116 +3,223 @@
+                )
+                region = response.get("LocationConstraint")
+        except ClientError:
+            logger.warn("Unable to fetch the region")
+
+        return region
+
+    def get_bucket_list(self):
+        """Returns bucket list from list_buckets response
+
+        Returns:
+            list: List of buckets
+        """
+        bucket_list = []
+        for bucket in self.bucket_list["Buckets"]:
+            bucket_list.append(bucket["Name"])
+        return bucket_list
 
     async def get_docs(self):
-        async with self.session.resource(
-            "s3", region_name=self.configuration["region"]
-        ) as s3:
-            bucket = await s3.Bucket(self.configuration["bucket"])
-            await asyncio.sleep(0)
-
-            async for obj_summary in bucket.objects.all():
-
-                doc_id = md5(obj_summary.key.encode("utf8")).hexdigest()
-                last_modified = await obj_summary.last_modified
-                await asyncio.sleep(0)
-
-                key = obj_summary.key
-                doc = {
-                    "_id": doc_id,
-                    "filename": key,
-                    "size": await obj_summary.size,
-                    "timestamp": last_modified.isoformat(),
-                }
-
-                async def _download(doc_id, key, timestamp=None, doit=None):
-                    if not doit:
-                        return
-
-                    # XXX check the checksum_crc32 of the file
-                    # obj = await obj_summary.Object()  # XXXX expensive
-                    # content_type = await obj.content_type
-
-                    if (
-                        # content_type not in SUPPORTED_CONTENT_TYPE
-                        os.path.splitext(key)[-1]
-                        not in SUPPORTED_FILETYPE
-                    ):
-                        logger.debug(f"{key} can't be extracted")
-                        return
-
-                    content = await self._get_content(key, timestamp)
-                    content["_id"] = doc_id
-                    return content
-
-                yield doc, partial(_download, doc_id, key)
+        """Get documents from Amazon S3
+
+        Returns:
+            dictionary: Document of file content
+
+        Yields:
+            dictionary: Document from Amazon S3.
+        """
+        bucket_list = self.configuration["buckets"] or self.get_bucket_list()
+        for bucket in bucket_list:
+            region_name = await self.get_bucket_region(bucket)
+            async with self.session.resource(
+                service_name="s3",
+                config=self.config,
+                endpoint_url=ENDPOINT_URL,
+                region_name=region_name,
+            ) as s3:
+                try:
+                    bucket_obj = await s3.Bucket(bucket)
+                    await asyncio.sleep(0)
+
+                    async for obj_summary in bucket_obj.objects.all():
```

Comment:
let's check if `all` loads in memory all docs on big buckets (that would do a memory bloat)

Reply:
@tarekziade we checked [aioboto3](https://github.com/terrycain/aioboto3/blob/main/aioboto3/resources/collection.py#:~:text=count%20%3D%200-,async,-for%20page%20in%20cast) libarary and it indirectly calls [pages](https://github.com/boto/boto3/blob/develop/boto3/resources/collection.py) method which returns a chunks of 1000 documents.
so `all` will not loads the complete data at a time in  memory.

Reply:
great! let's surface the page size as an option if we want to tweak this